libevmasm/SemanticInformation.cpp

```diff
@@ -202,6 +202,27 @@ bool SemanticInformation::movable(Instruction _instruction)
 	return true;
 }
 
+bool SemanticInformation::sideEffectFree(Instruction _instruction)
+{
+	// These are not really functional.
+	if (isDupInstruction(_instruction) || isSwapInstruction(_instruction))
```

Comment:
Should we rather assert here instead of returning ``false``? Considering these "not side-effect free" might be a bit misleading... We shouldn't be calling this for ``DUP`` or ``SWAP`` anyways, should we?

Reply:
I think the biggest problem is that there is no fitting kind of assert to use here ;)

Reply:
Hm... ``assertThrow(!isDupInstruction(_instruction) && !isSwapInstruction(_instruction), AssemblyException, "");``? That's the closest I see short of adding a new one - but maybe it's not that important in the end :-).

Reply:
yeah, AssemblyException could do...